server/swap/swap.go

```diff
@@ -1033,17 +1060,17 @@ func (s *Swapper) checkInactionEventBased() {
 
 		log.Tracef("checkInactionEventBased: match %v (%v)", match.ID(), match.Status)
 
-		failMatch := func(fault bool) {
-			s.failMatch(match, fault)
-			deletions = append(deletions, match)
+		deleteMatch := func(fault bool) {
+			s.deleteMatch(match)
+			failures = append(failures, fail{match, fault}) // to process after map delete
 		}
 
 		switch match.Status {
```

Comment:
while reading this code, noticed [fallthrough](https://github.com/LasTshaMAN/dcrdex/blob/8c190185d64795db9b698f20cccfe004dc8e6d13/server/swap/swap.go#L1082-L1084) usage here that seems a bit awkward to me,

although these checks (for [Maker](https://github.com/LasTshaMAN/dcrdex/blob/8c190185d64795db9b698f20cccfe004dc8e6d13/server/swap/swap.go#L1077-L1077), for [Taker](https://github.com/LasTshaMAN/dcrdex/blob/8c190185d64795db9b698f20cccfe004dc8e6d13/server/swap/swap.go#L1088-L1088)) are probably not mutually exclusive, [deleteMatch](https://github.com/LasTshaMAN/dcrdex/blob/8c190185d64795db9b698f20cccfe004dc8e6d13/server/swap/swap.go#L349-L349) might be called 2 times (because of that `fallthrough`),
I've reproduces this in tests (just add 2nd call to [deleteMatch here](https://github.com/LasTshaMAN/dcrdex/blob/8c190185d64795db9b698f20cccfe004dc8e6d13/server/swap/swap_test.go#L2144-L2144) and add panic under [corresponding if](https://github.com/LasTshaMAN/dcrdex/blob/8c190185d64795db9b698f20cccfe004dc8e6d13/server/swap/swap.go#L363-L363))

I don't have enough context in this area at this point, but maybe you'd wanna take a look

Reply:
Fair enough, it's confusing so worth tweaking.

Although, `match.expiredBy` checks if *either* contract is expired, so when calling that after `fallthrough` **in `order.MakerSwapCast` status**, that is checking only the maker's existing contract expiry, not taker's since taker's does not exist yet.

Because by definition the taker's contract MUST expire before maker's (by 12 hours on mainnet), the only way in practice it would failMatch twice is if dcrdex were suspended or shutdown for a long time and then restarted, only to find BOTH the maker's existing contract is expired (20 hours old) and the expected lock time of the non-existent taker contract would be expired.  So in practice it won't happen.  For the same reason, there's really no need to check the maker's own contract for expiry since as the comment above the `fallthrough` explains and I've stated, `expectedTakerLockTime` must be well before maker's contract expiry.

This was an overzealous attempt at reducing code duplication (calling `match.expiredBy` explicitly in this `case` instead of falling through).  I'll either put a `break` after the first `failMatch`, or simply remove the `fallthrough` since as I said the `match.expiredBy` check cannot hit if the `expectedTakerLockTime` did not.

Reply:
I see, thank you, the latest commit reads much nicer!